### ISSUE TITLE
Ability to edit multiple lines that share the same selected substring

### ISSUE THREAD

**Issue by bnovak331** [YoYoGames/GameMaker-Feature-Requests#1212]
### Is your feature request related to a problem?

Sometimes in my GML scripts I may have a lot of variables or logic that is SIMILAR enough in their name, which makes it frustrating when I want to make changes, but not necessarily "Replace All".

For example: suppose I have a few different stat types on a character:
stat_health
stat_attack
stat_defense
etc.

If I want to do away with the names of "health" "attack" and "defense", I have to go one-by one. I can't select all instances of "stat_", and then delete the trailing text.

### Describe the solution you'd like

Visual Studio Code has the exact feature that I think would be very nice to have in GML.

Taking the example above, if I select "stat_" somewhere in the script on VS Code, and press Ctrl+D (on Windows), it will also select the very next instance in the code where "stat_" is typed out.

Then I have two selections - if I type something, it will make the change in both places of the code (different from replace all). If I keep pressing Ctrl+D, it will select the THIRD instance where "stat_" is typed, and so on.

I've started counting the times this would be very handy to have enough to where I felt it was worth putting out a feature request.

**Comment by gnysek** [comment]
Alt+mouse click/drag:

![obraz](https://github.com/YoYoGames/GameMaker-Feature-Requests/assets/524094/22c99655-b110-4cd1-b284-06e855f9c296)

**Comment by bnovak331** [comment]
Ah, I see. What about cases like this, where the keywords might not be
fully aligned?
[image: image.png]

The "replace" feature is obviously a solution here. I just think a hotkey
would be nice to have (unless I'm missing one), something similar to the VS
Code shortcut.
[image: ctrlD.gif]

**Comment by bnovak331** [comment]
Ah, I see. What about cases like this, where the keywords might not be
fully aligned?
![image](https://github.com/YoYoGames/GameMaker-Feature-Requests/assets/75385894/9d256eac-c117-4efd-9cb3-5ea90b418f66)

The "replace" feature is obviously a solution here. I just think a hotkey
would be nice to have (unless I'm missing one), something similar to the VS
Code shortcut.
![ctrlD](https://github.com/YoYoGames/GameMaker-Feature-Requests/assets/75385894/5cc8e6ef-dcf0-4d1f-bf05-83362b7ec558)

**Comment by gnysek** [comment]
Yes, so that's feature I've seen in some editors (ability to select multiple words with different length in different places) and that's what's missing here, as GM only allows same width (and same columns only, with no gaps).
